### ISSUE TITLE
Results with Combined Noise lower than expected

### ISSUE THREAD

**Issue by pierrefdz** [ando-khachatryan/HiDDeN#24]
Hi and thanks for the code and architecture! 
I was wondering why the results with the combined noise are lower than the ones from the paper. Indeed, the original paper recovers 100% accuracy for the combined models when dealing with untransformed watermarked images, while we only obtain around 90% for networks trained with combined noise here. 
Do you happen to know why ?

**Comment by ando-khachatryan** [comment]
Hi, 
The noise layer implementation is a bit different (more general) from the paper. You can obtain the effect of the paper noise layers but you'd need to tweak with the noise layer parameters. See the description of noise layers and compare to the paper version for details.

**Comment by pierrefdz** [comment]
Hi and thanks for the quick answer.
I've tried with dropout(0.3,0.3)+dropout(0.7,0.7)+cropout((0.84,0.84),(0.84,0.84))+cropout((0.55,0.55),(0.55,0.55))+crop((0.84,0.84),(0.84,0.84))+crop((0.55,0.55),(0.55,0.55))+blur(11,11)+blur(25,25)+jpeg() as noise, which should correspond to what is used in the paper, but I've obtained a BER of 0.1.
However, I ran the experiment for more epochs and it seems that there is no convergence at 400 epochs, the issue might come from the fact that more iterations are needed than 400.

**Comment by SVithurabiman** [comment]
Hi @pierrefdz  and @ando-khachatryan,

Thank you for your insightful discussion post.

I have attempted to train the model for `600 epochs` using the noise configuration mentioned:
`dropout(0.3,0.3) + dropout(0.7,0.7) + cropout((0.84,0.84),(0.84,0.84)) + cropout((0.55,0.55),(0.55,0.55)) + crop((0.84,0.84),(0.84,0.84)) + crop((0.55,0.55),(0.55,0.55)) + blur(11,11) + blur(25,25) + jpeg()`.

After testing on 3000 COCO images, I observed an average BER of approximately 0.15. I also evaluated the model at various checkpoints (e.g., 300, 400, and 600 epochs) but obtained similar results.

Could you please share any insights on whether a lower BER was achieved, if so, how such a low BER was obtained?

**Comment by pierrefdz** [comment]
> Hi @pierrefdz  and @ando-khachatryan,
> 
> Thank you for your insightful discussion post.
> 
> I have attempted to train the model for `600 epochs` using the noise configuration mentioned:
> `dropout(0.3,0.3) + dropout(0.7,0.7) + cropout((0.84,0.84),(0.84,0.84)) + cropout((0.55,0.55),(0.55,0.55)) + crop((0.84,0.84),(0.84,0.84)) + crop((0.55,0.55),(0.55,0.55)) + blur(11,11) + blur(25,25) + jpeg()`.
> 
> After testing on 3000 COCO images, I observed an average BER of approximately 0.15. I also evaluated the model at various checkpoints (e.g., 300, 400, and 600 epochs) but obtained similar results.
> 
> In the paper “_Watermarking Images in Self-Supervised Latent-Spaces_” authored by @pierrefdz, the reported BER is 0.00. Could you please share any insights on how this was achieved or how such a low BER was obtained?

In the SSL watermarking paper I just copied the results from the HiDDeN's paper table, I think this is mentioned somewhere in the text. 
I also think that in HiDDeN the bit accuracy on the combined noise is not very high.

On my side I've tried to reproduce the results from this paper for a long time, but I never managed to get something very close to it with the same number of training steps.

**Comment by SVithurabiman** [comment]
> > Hi [@pierrefdz](https://github.com/pierrefdz)  and [@ando-khachatryan](https://github.com/ando-khachatryan),
> > Thank you for your insightful discussion post.
> > I have attempted to train the model for `600 epochs` using the noise configuration mentioned:
> > `dropout(0.3,0.3) + dropout(0.7,0.7) + cropout((0.84,0.84),(0.84,0.84)) + cropout((0.55,0.55),(0.55,0.55)) + crop((0.84,0.84),(0.84,0.84)) + crop((0.55,0.55),(0.55,0.55)) + blur(11,11) + blur(25,25) + jpeg()`.
> > After testing on 3000 COCO images, I observed an average BER of approximately 0.15. I also evaluated the model at various checkpoints (e.g., 300, 400, and 600 epochs) but obtained similar results.
> > In the paper “_Watermarking Images in Self-Supervised Latent-Spaces_” authored by [@pierrefdz](https://github.com/pierrefdz), the reported BER is 0.00. Could you please share any insights on how this was achieved or how such a low BER was obtained?
> 
> In the SSL watermarking paper I just copied the results from the HiDDeN's paper table, I think this is mentioned somewhere in the text. I also think that in HiDDeN the bit accuracy on the combined noise is not very high.
> 
> On my side I've tried to reproduce the results from this paper for a long time, but I never managed to get something very close to it with the same number of training steps.

 Thank you for your prompt response. I just corrected and edited the previous comment after going through your paper once again. 
As an additional note, there was a slight increase in robustness when it was trained longer. However, it did not match the published results.
